# Лабораторная работа 1
Вариант 5 - Тищенко Т.Д.
гр. ПИН-231

Импортируем необходимые библиотеки

In [ ]:
import pandas as pd
import numpy as np

Загружаем файлы auto1 и auto2 поместив их в соотвествующие датафремы

In [ ]:
auto1 = pd.read_csv('auto1.csv')
auto2 = pd.read_csv('auto2.csv')

Поочередно выводим содержимое первых 10 строк каждого датафрейма.

In [ ]:
print(auto1.head(10))

     mpg  cylinders  displacement horsepower  weight  acceleration  model  \
0   18.0          8           307        130    3504          12.0     70   
1   15.0          8           350        165    3693          11.5     70   
2   18.0          8           318        150    3436          11.0     70   
3   16.0          8           304        150    3433          12.0     70   
4   17.0          8           302        140    3449          10.5     70   
5   15.0          8           429        198    4341          10.0     70   
6   14.0          8           454        220    4354           9.0     70   
7   14.0          8           440        215    4312           8.5     70   
8   14.0          8           455        225    4425          10.0     70   
9   15.0          8           390        190    3850           8.5     70   
10  15.0          8           383        170    3563          10.0     70   
11  14.0          8           340        160    3609           8.0     70   

In [ ]:
print(auto2.head(10))

     mpg  cylinders  displacement horsepower  weight  acceleration  model  \
0   33.0          4            91         53    1795          17.4     76   
1   20.0          6           225        100    3651          17.7     76   
2   18.0          6           250         78    3574          21.0     76   
3   18.5          6           250        110    3645          16.2     76   
4   17.5          6           258         95    3193          17.8     76   
5   29.5          4            97         71    1825          12.2     76   
6   32.0          4            85         70    1990          17.0     76   
7   28.0          4            97         75    2155          16.4     76   
8   26.5          4           140         72    2565          13.6     76   
9   20.0          4           130        102    3150          15.7     76   
10  13.0          8           318        150    3940          13.2     76   
11  19.0          4           120         88    3270          21.9     76   

Выводим информачию о количестве строк и столбцов датафреймов auto1 и auto2
из которых видно, что первый содержит 13 столбцов и 198 строк, а второй 8 столбцов и 200 строк, соответственно.

In [ ]:
auto1.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 198 entries, 0 to 197
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           198 non-null    float64
 1   cylinders     198 non-null    int64  
 2   displacement  198 non-null    int64  
 3   horsepower    198 non-null    object 
 4   weight        198 non-null    int64  
 5   acceleration  198 non-null    float64
 6   model         198 non-null    int64  
 7   origin        198 non-null    int64  
 8   car           198 non-null    object 
 9   Unnamed: 9    0 non-null      float64
 10  Unnamed: 10   0 non-null      float64
 11  Unnamed: 11   0 non-null      float64
 12  Unnamed: 12   0 non-null      float64
 13  Unnamed: 13   0 non-null      float64
dtypes: float64(7), int64(5), object(2)
memory usage: 21.8+ KB


In [ ]:
auto2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           200 non-null    float64
 1   cylinders     200 non-null    int64  
 2   displacement  200 non-null    int64  
 3   horsepower    200 non-null    object 
 4   weight        200 non-null    int64  
 5   acceleration  200 non-null    float64
 6   model         200 non-null    int64  
 7   origin        200 non-null    int64  
 8   car           200 non-null    object 
dtypes: float64(2), int64(5), object(2)
memory usage: 14.2+ KB


Выводим индексы датафреймов auto1 и auto2, соответственно.

In [ ]:
auto1.index

RangeIndex(start=0, stop=198, step=1)

In [ ]:
auto2.index

RangeIndex(start=0, stop=200, step=1)

Создаем копию датафрейма auto1 не содержащую нулевых Unnamed столбцов

In [ ]:
auto1_copy = auto1.copy()
cut_coll = ['Unnamed: 9', 'Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13']
auto1_copy.drop(cut_coll, axis = 1, inplace = True)
auto1_copy.head()

,mpg,cylinders,displacement,horsepower,weight,acceleration,model,origin,car
0,18.0,8,307,130,3504,12.0,70,1,chevrolet chevelle malibu
1,15.0,8,350,165,3693,11.5,70,1,buick skylark 320
2,18.0,8,318,150,3436,11.0,70,1,plymouth satellite
3,16.0,8,304,150,3433,12.0,70,1,amc rebel sst
4,17.0,8,302,140,3449,10.5,70,1,ford torino


Объединяем датафремы путем конкатинации.

In [ ]:
auto = pd.concat([auto1_copy, auto2], ignore_index=True)
print(auto.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-null    int64  
 2   displacement  398 non-null    int64  
 3   horsepower    398 non-null    object 
 4   weight        398 non-null    int64  
 5   acceleration  398 non-null    float64
 6   model         398 non-null    int64  
 7   origin        398 non-null    int64  
 8   car           398 non-null    object 
dtypes: float64(2), int64(5), object(2)
memory usage: 28.1+ KB
None


Добавляем столбец 'owners' с рандомными значениями
в диапазоне от 10000 до 60000

In [ ]:
auto['owners'] = np.random.randint(10000, 60001, size=len(auto))
print(auto.head())
print(auto.info())

    mpg  cylinders  displacement horsepower  weight  acceleration  model  \
0  18.0          8           307        130    3504          12.0     70   
1  15.0          8           350        165    3693          11.5     70   
2  18.0          8           318        150    3436          11.0     70   
3  16.0          8           304        150    3433          12.0     70   
4  17.0          8           302        140    3449          10.5     70   

   origin                        car  owners  
0       1  chevrolet chevelle malibu   37839  
1       1          buick skylark 320   18879  
2       1         plymouth satellite   17967  
3       1              amc rebel sst   19811  
4       1                ford torino   39088  
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 398 entries, 0 to 397
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   mpg           398 non-null    float64
 1   cylinders     398 non-

Определяем временной диапазон выпуска можелей датафрейма путем определения максимального и минимального значений столбца 'model'

In [ ]:
print(f'Ranged "model" {auto['model'].min()} - {auto['model'].max()}')

Ranged "model" 70 - 82


Определяем как менялся расход автомобилей в течении данного периода, группируя средние значения расзхода за каждый год периода

In [ ]:
auto.groupby('model')['mpg'].mean()

,mpg
model,
70,17.689655
71,21.250000
72,18.714286
73,17.100000
74,22.703704
75,20.266667
76,21.573529
77,23.375000
78,24.061111


Выводим "Топ 5" предпочтительных моделей автомобилей за всё время

In [ ]:
auto['car'].value_counts().head(5)

,count
car,
ford pinto,6
ford maverick,5
amc matador,5
toyota corolla,5
amc hornet,4


Проверяем зависимость расхода от количнства целиндров группируя средние значеия расхода по количеству целиндров автомобилей.

In [ ]:
auto.groupby('cylinders')['mpg'].mean()

,mpg
cylinders,
3,20.550000
4,29.286765
5,27.366667
6,19.985714
8,14.963107


Докажем зависимость расхода автомобиля от веса, вычислив кореляцию значений.

результат "-0.832" показывает сильную обратную зависмость

In [ ]:
print(f"Корреляция: {auto['mpg'].corr(auto['weight']):.3f}")
print(f'\n {auto.groupby('weight')['mpg'].mean()}')

Корреляция: -0.832

 weight
1613    35.0
1649    31.0
1755    39.1
1760    35.1
1773    31.0
        ... 
4951    12.0
4952    12.0
4955    12.0
4997    11.0
5140    13.0
Name: mpg, Length: 351, dtype: float64


Сортируем датафрейм по значению столбца 'weight' по убыванию

In [ ]:
auto.sort_values(by = 'weight', ascending = False)

,mpg,cylinders,displacement,horsepower,weight,acceleration,model,origin,car,owners
44,13.0,8,400,175,5140,12.0,71,1,pontiac safari (sw),27395
103,11.0,8,400,150,4997,14.0,73,1,chevrolet impala,46269
42,12.0,8,383,180,4955,11.5,71,1,dodge monaco (sw),52136
90,12.0,8,429,198,4952,11.5,73,1,mercury marquis brougham,31116
95,12.0,8,455,225,4951,11.0,73,1,buick electra 225 custom,52975
...,...,...,...,...,...,...,...,...,...,...
53,31.0,4,71,65,1773,19.0,71,3,toyota corolla 1200,44155
345,35.1,4,81,60,1760,16.1,81,3,honda civic 1300,57983
343,39.1,4,79,58,1755,16.9,81,3,toyota starlet,14180
144,31.0,4,76,52,1649,16.5,74,3,toyota corona,31244
